### PR TITLE
[AutoParallel] saved tensor hook support auto parallel

### PR DESCRIPTION
**Pull request by wanghuancoder**
### PR types

Others 
### PR changes

Others 
### Description

saved tensor hook支持AutoParallel

Pcard-73145

### CODE REVIEW

File: paddle/fluid/eager/tensor_wrapper.h [hunk 1]
```diff
@@ -95,6 +95,25 @@ class TensorWrapper {
...
         packed_value_ = (*pack_hook)(tensor);
+      } else if (egr::SavedTensorsHooks::GetInstance().IsEnable() &&
+                 tensor.is_dist_tensor() && tensor.initialized()) {
+        intermidiate_tensor_.set_impl(
+            std::make_shared<phi::distributed::DistTensor>(
+                tensor.dims(),
+                static_cast<phi::distributed::DistTensor*>(tensor.impl().get())
+                    ->dist_attr()));
+        phi::DenseTensor* dense_tensor =
+            static_cast<phi::distributed::DistTensor*>(tensor.impl().get())
```
Comment (GhostScreaming) [inline paddle/fluid/eager/tensor_wrapper.h:106]: 这里是不是用`const phi::DenseTensor& dense_tensor = static_cast<phi::distributed::DistTensor*>(tensor.impl().get())->value()`就好了

Comment (wanghuancoder) [reply paddle/fluid/eager/tensor_wrapper.h:106]: 好的修改了 谢谢

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_semi_auto_parallel_saved_tensor_hook
                       PROPERTIES LABELS "RUN_TYPE=EXCLUSIVE" TIMEOUT 100)`

**Review by GhostScreaming — APPROVED** [review]
LGTM

**Pull request merged**